app/vmagent/remotewrite/remotewrite.go

```diff
@@ -103,7 +103,7 @@ var (
 	streamAggrDropInputLabels = flagutil.NewArrayString("streamAggr.dropInputLabels", "An optional list of labels to drop from samples "+
 		"before stream de-duplication and aggregation . See https://docs.victoriametrics.com/stream-aggregation.html#dropping-unneeded-labels")
 
-	disableOnDiskQueue = flag.Bool("remoteWrite.disableOnDiskQueue", false, "Whether to disable storing pending data to -remoteWrite.tmpDataPath "+
+	disableOnDiskQueuePerURL = flagutil.NewArrayBool("remoteWrite.disableOnDiskQueue", "Whether to disable storing pending data to -remoteWrite.tmpDataPath "+
```

Comment:
I don't think, that it's possible to disable it for some queue. It makes ingestion results inconsistent and may lead to data loss.
 Main purpose of having persistent queue disabled - build data pipeline with a chain of vmagents.

 I'd suggest to leave this flag as is.

Reply:
Thank you for your feedback.

In our setup, critical metrics require disk persistence for reliability during network outages. However we don't want to persist our high cardinality metrics that are sent in a cluster without strong guarantees so we have all the disks available for the metrics we really need to keep. The ingestion would indeed be inconsistent between remote writes, its the purpose of this proposal :smiley:.

Reply:
As workaround, it's possible to configure max queue size per remoteWrite target with: 
-maxPendingBytesPerURL=100MB,100GB 
-remoteWrite.URL=http://not-important-dst,http://important-dst

Reply:
Yes, it'll add additional disk pressure. 

Thanks for the explanation, this change makes sense to me.